src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestCallback.cs

```diff
@@ -305,6 +305,16 @@ private static void OnRequestError(WinHttpRequestState state, Interop.WinHttp.WI
             
             Debug.Assert(state != null, "OnRequestError: state is null");
 
+            // Add instrumentation for now to catch unexpected errors and fail fast and produce a crash dump.
```

Comment:
> Any reason not to add them everywhere we think WinHTTP returns this?

My previous is PR is like experiment and I'm not sure which place could go wrong, so I added many things and try to be inclusive. There was discussion offline and eliminate some options which are not likely. I am not 100 percent sure about where to place the Debug.Assert statement due to limited information we currently have, but I want to merge this PR and observe what will happen so that I can have more information to move forward.